packages/std/src/math/uint256.rs

```diff
@@ -0,0 +1,560 @@
+        Self(self.0.saturating_sub(other.0))
+    }
+
+    pub fn saturating_mul(self, other: Self) -> Self {
+        Self(self.0.saturating_mul(other.0))
+    }
+}
+
+impl From<u128> for Uint256 {
+    fn from(val: u128) -> Self {
+        Uint256(val.into())
+    }
+}
+
+impl From<u64> for Uint256 {
+    fn from(val: u64) -> Self {
+        Uint256(val.into())
+    }
+}
+
+impl From<u32> for Uint256 {
+    fn from(val: u32) -> Self {
+        Uint256(val.into())
+    }
+}
+
+impl From<u16> for Uint256 {
+    fn from(val: u16) -> Self {
+        Uint256(val.into())
+    }
+}
+
+impl From<u8> for Uint256 {
+    fn from(val: u8) -> Self {
+        Uint256(val.into())
+    }
+}
+
+impl TryFrom<&str> for Uint256 {
+    type Error = StdError;
+
+    fn try_from(val: &str) -> Result<Self, Self::Error> {
+        match val.parse::<U256>() {
+            Ok(u) => Ok(Uint256(u)),
+            Err(e) => Err(StdError::generic_err(format!("Parsing u256: {}", e))),
+        }
+    }
+}
+
+impl From<Uint256> for String {
+    fn from(original: Uint256) -> Self {
+        original.to_string()
+    }
+}
+
+impl fmt::Display for Uint256 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl ops::Add<Uint256> for Uint256 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Uint256(self.0.checked_add(rhs.0).unwrap())
+    }
+}
+
+impl<'a> ops::Add<&'a Uint256> for Uint256 {
+    type Output = Self;
+
+    fn add(self, rhs: &'a Uint256) -> Self {
+        Uint256(self.0.checked_add(rhs.0).unwrap())
+    }
+}
+
+impl ops::Sub<Uint256> for Uint256 {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        Uint256(self.0.checked_sub(rhs.0).unwrap())
+    }
+}
+
+impl<'a> ops::Sub<&'a Uint256> for Uint256 {
+    type Output = Self;
+
+    fn sub(self, rhs: &'a Uint256) -> Self {
+        Uint256(self.0.checked_sub(rhs.0).unwrap())
+    }
+}
+
+impl ops::Div<Uint256> for Uint256 {
+    type Output = Self;
+
+    fn div(self, rhs: Self) -> Self::Output {
+        Self(self.0.checked_div(rhs.0).unwrap())
+    }
+}
+
+impl<'a> ops::Div<&'a Uint256> for Uint256 {
+    type Output = Self;
+
+    fn div(self, rhs: &'a Uint256) -> Self::Output {
+        Self(self.0.checked_div(rhs.0).unwrap())
+    }
+}
+
+impl ops::Shr<u32> for Uint256 {
+    type Output = Self;
+
+    fn shr(self, rhs: u32) -> Self::Output {
+        if rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        Self(self.0.shr(rhs))
+    }
+}
+
+impl<'a> ops::Shr<&'a u32> for Uint256 {
+    type Output = Self;
+
+    fn shr(self, rhs: &'a u32) -> Self::Output {
+        if *rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        Self(self.0.shr(*rhs))
```

Comment:
Can we just do this?

```suggestion
        self.shr(*rhs)
```

Reply:
Yeah, good point.